packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -207,52 +203,216 @@ function useMediaUploadQueue() {
+  const muteVideoItem = useCallback(
+    async (item) => {
+      const { id, file, additionalData = {} } = item;
+
+      startMuting({ id });
+
+      try {
+        const newFile = await stripAudioFromVideo(file);
+
+        if (!isMounted.current) {
+          return;
+        }
+
+        additionalData.web_stories_is_muted = true;
+        finishMuting({ id, file: newFile, additionalData });
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startMuting, finishMuting, stripAudioFromVideo, cancelUploading]
+  );
+
+  const optimizeVideoItem = useCallback(
+    async (item) => {
+      const { id, file, additionalData = {} } = item;
+
+      startTranscoding({ id });
+
+      try {
+        const newFile = await transcodeVideo(file);
+
+        if (!isMounted.current) {
+          return;
+        }
+
+        additionalData.web_stories_media_source = 'video-optimization';
+        finishTranscoding({ id, file: newFile, additionalData });
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        currentTranscodingItem.current = null;
+      }
+    },
+    [startTranscoding, finishTranscoding, transcodeVideo, cancelUploading]
+  );
+
+  const uploadItem = useCallback(
+    async (item) => {
+      const { id, file, resource, additionalData = {} } = item;
+      let { posterFile } = item;
+
+      let newResource;
+
+      const posterFileName = getFileName(file) + '-poster.jpeg';
+
+      startUploading({ id });
+
+      // If we don't have a poster yet (e.g. after converting a GIF),
+      // try to generate one now.
+      if (!resource.poster && !posterFile) {
+        try {
+          posterFile = await getFirstFrameOfVideo(file);
+        } catch {
+          // Not interested in errors here.
+        }
+      }
+
+      trackEvent('upload_media', {
+        file_size: file?.size,
+        file_type: file?.type,
+      });
+
+      //eslint-disable-next-line @wordpress/no-unused-vars-before-return
+      const trackTiming = getTimeTracker('load_upload_media');
+
+      try {
+        // The newly uploaded file won't have a poster yet.
+        // However, we'll likely still have one on file.
+        // Add it back so we're never without one.
+        // The final poster will be uploaded later by uploadVideoPoster().
+        newResource = await uploadFile(file, additionalData);
+
+        if (!isMounted.current) {
+          return;
+        }
+      } catch (error) {
+        // Cancel uploading if there were any errors.
+        cancelUploading({ id, error });
+
+        trackError('upload_media', error?.message);
+      } finally {
+        trackTiming();
+      }
+
       try {
```

Comment:
I would see poster upload to move to it's own effect. It could even have it's own state. So
```
UPLOADING -> UPLOADED -> FINISHED
```
Becomes 
```
UPLOADING -> UPLOADED -> UPLOADING-POSTER-> FINISHED
```

Thoughts.

Reply:
Hmm not sure it warrants its own separate status 🤔  especially since it only applies to videos.

What I like about the current way is we only consider a video to be fully uploaded once we have at least attempted to upload a poster.

That said, `uploadItem` definitely could need some tweaking, because right now it looks like it tries to upload a poster even for images 🤔